go.mod

```diff
@@ -1,32 +1,37 @@
 module go.temporal.io/sdk
 
-go 1.20
+go 1.21
```

Comment:
Hrmm, do we want to bump minimum Go version? Not that I mind, just want to make sure it's a conscious decision and not buried in this PR

Reply:
I'm not sure why this happened, maybe because the nexus SDK depends on `slog`?

Reply:
Yeah, or more importantly, Nexus Go SDK itself requires 1.21+ which infects all users of it. Technically that's the oldest non-EOL, @Quinn-With-Two-Ns - thoughts?

Also, another concern I now have is about transitive dependencies. Nexus SDK is now forcing `github.com/gorilla/mux` to at least `v1.8.0` for every user of the Temporal Go SDK. I am not sure we want to force this dependency or our needed version of it on every user.

Reply:
I'm sure we can do without this dependency. I know Go's built-in router has advanced in recent versions.
Can we live with this for the immediate future and investigate next week?

Reply:
Yeah `github.com/gorilla/mux` is quite a rough dependency to start requiring for non nexus users.

Reply:
Upgrading to `go 1.21` is fine if it is needed